### DISCUSSION TITLE
Teams can't share a server?

### DISCUSSION THREAD

**Discussion by bitnom** [coollabsio/coolify#1820] [Q&A] [answered]
This question has been asked many times on Discord without being answered. I install locally (1 server), create a new team, and I cannot give the team access to the server. Please tell me if it can somehow be done without breaking anything. If not, it needs to be a feature request (Multiple teams using the same server(s))

**Comment by maaaathis** [comment]
I also find it really confusing, especially as this option exists in the Sources. There it is a "System Wide?" checkbox.

**Comment by dawnniie** [comment]
A 'system wide' option for servers (or even just a way to clone) would be incredibly useful, at the moment for me it's just totally unrealistic to copy over keys and configurations for all my different servers. I can't use teams at all right now for this exact reason. Please make this easier!

**Comment by nsrau** [comment]
Does anyone know how to solve the problem of using localhost for more than one team?

**Reply by gmrizzo** [reply]
I don't think that this would be possible right now. If you setup the same server on another team, it would try to setup a second proxy as of my understanding. Two cannot run in parallel with the same configurations.

**Reply by nsrau** [reply]
ok, but what I don't understand is why it isn't possible to just hide projects between teams for example, but keep the same settings needed for the minimum to work.

**Reply by luckydonald** [reply]
My current work around is just adding the same server (i.e. localhost) to the other team under the server section as well.
Server may need a restart physically, because it doesn't seem to expect the docker to change from the install script, but after that it seems to work well enough.

**Comment by gabrielcossette** [comment]
I agree this makes the permission model much less useful.

However, this issue added to 4.1 milestone would solve the issue it seems?

https://github.com/coollabsio/coolify/issues/2378

**Comment by luckydonald** [comment]
My current **work around** is just adding the same server (i.e. localhost) to the other team under the server section as well.
Server may need a restart physically, because it doesn't seem to expect the docker to change from the install script, but after that it seems to work well enough.

---

**Update 2026-03-14:**

If you get an error `the server with domain… is already in use by a different team`, try to add it under a different name:
- for `localhost`, you may try `127.0.0.1` for IPv4 or `::1` for IPv6 as alternative, maybe this (new?) duplicate detection is dumb enough to consider it as a new host instead, if the strings don't match.
- If you use a domain, try adding it with a different (sub-)domain alias (Second A or AAAA record pointing to the same IP) or a raw IP address instead of the previously used domain.

**Reply by rubslopes** [reply]
> My current work around is just adding the same server (i.e. localhost) to the other team under the server section as well. Server may need a restart physically, because it doesn't seem to expect the docker to change from the install script, but after that it seems to work well enough.

Hi, I'm trying to reproduce your solution. I noticed that it creates two proxies containers. Has this caused problems to you in anyway?

**Reply by enesbala5** [reply]
@rubslopes @luckydonald Any update on this? Did it cause issues for you? I can't access services on the secondary team right now at all

**Reply by luckydonald** [reply]
So far no problems. Took a couple attempts to get it started initially (web interface will be gone for a minute, the it will cry for about 5 minutes and I spam restart), but it's been running since.

**Reply by endquote** [reply]
I tried to do this but it insists on me selecting a private key, and there's none to select from.

**Reply by luckydonald** [reply]
@endquote
Yeah, because it's a different team. Obviously they wouldn't automatically share login credentials.

Just generate a new one, and add it to authorized_keys like you normally would.
Or copy the root private key over.

**Reply by luckydonald** [reply]
If you get an error `the server with domain… is already in use by a different team`, try to add it under a different name:
- for `localhost`, you may try `127.0.0.1` for IPv4 or `::1` for IPv6 as alternative, maybe this (new?) duplicate detection is dumb enough.
- If it's a domain, try adding it with a different (sub-)domain or a raw IP address instead.

**Comment by rubslopes** [comment]
Me too. No problems so far.

Em sáb., 12 de abr. de 2025, 09:30, Luckydonald ***@***.***>
escreveu:

**Comment by derSato** [comment]
to be fair I don't want to run the same traefik + crowdsec + appsec + rate limiter on the new teams proxy again since its somewhat taxing on the system.
Like people point out hiding projects would be all that's needed.

**Reply by luckydonald** [reply]
Yeah, having an option to invite to simply share an existing server with a existing team would be way better.
Registering a server twice is only an ugly workaround until that's implemented.

**Comment by Default-01** [comment]
We need this, and it should be added asap!

**Comment by gauel** [comment]
Any update on this?

**Reply by luckydonald** [reply]
https://github.com/coollabsio/coolify/discussions/1820#discussioncomment-12200360 still works, that's all we know. My guess is that they are very busy with the coolify rewrite #5685.

**Comment by nicanordlc** [comment]
🙏🙏🙏 It would be good to divide group of projects by team on one big instance/server.

**Comment by yipfram** [comment]
Maybe by using different docker networks you can make this more secure?

Because sharing the server means everyone has access to root via terminal section... Not very safe :)

**Reply by luckydonald** [reply]
Probably a docker proxy could help?
I had no need as I trust my team.

**Comment by amirhmoradi** [comment]
Hi, thanks for the great work on Coolify. 
I would love to see this feature implemented or at least workarounds documented please.

**Comment by endquote** [comment]
I swear that in an earlier version, I was able to have multiple teams all using localhost. I think this changed at some point.

Also need the ability to move things between teams that are using the same server.

**Reply by luckydonald** [reply]
First part is still possible: https://github.com/coollabsio/coolify/discussions/1820#discussioncomment-12200360

**Comment by maukoese** [comment]
When I try to add the same server to another team I get an error about "the server with domain is already in use by a different team", so unable to use this workaround.

**Reply by luckydonald** [reply]
If it's localhost, you may try 127.0.0.1 for IPv4 or ::1 for IPv6 as alternative, maybe this (new?) duplicate detection is dumb enough to think it's a "different" server.
If you use a domain, try adding it with a different (sub-)domain or a raw IP address instead.

**Comment by Cinzya** [comment, marked as answer]
Just to clarify some things here, as this topic comes up more often now:
Teams are not supposed to share the same server as of this moment. The main reason is, there need to be more safeguards put in place, to ensure Team A can't access and alter the deployments of Team B, when both use the same server. **Coolify does not have those safeguards yet**, even if it might appear so on first glance.

Even if you disable the Terminal for the Team, nothing stops a user to mount the directory / volume of an App from the other team and make changes to this App.

Workarounds are at the moment being patched, as Coolify v4's stable release is being prepared. It is unclear, when / if this will be made possible again for v4, but it is on the roadmap for Coolify v5. Until then, we do not recommend trying to make two teams share the same server-

**Reply by maukoese** [reply]
Thank you for the heads up. Our use case was same VPS handling some 5 projects from 3 different Github organizations and would have preferred them to be under different teams - but of course can't deploy with that setup.

I suppose we can have all of them on one team for now and look at moving in the future.

**Reply by b2balexit** [reply]
> Just to clarify some things here, as this topic comes up more often now: Teams are not supposed to share the same server as of this moment. The main reason is, there need to be more safeguards put in place, to ensure Team A can't access and alter the deployments of Team B, when both use the same server. **Coolify does not have those safeguards yet**, even if it might appear so on first glance.
> 
> Even if you disable the Terminal for the Team, nothing stops a user to mount the directory / volume of an App from the other team and make changes to this App.
> 
> Workarounds are at the moment being patched, as Coolify v4's stable release is being prepared. It is unclear, when / if this will be made possible again for v4, but it is on the roadmap for Coolify v5. Until then, we do not recommend trying to make two teams share the same server-

Do you have any news about this?

**Comment by king3465** [comment]
hi
It depends on how the server is configured and what resources are available. Many teams can share a server successfully when proper access controls, monitoring, and resource allocation are in place. Performance and security should always be priorities. I’ve come across discussions on[ https://astutebetaserveri.com/]( https://astutebetaserveri.com/) that highlight how reliable server management can help multiple users work efficiently without unnecessary conflicts.

**Comment by julianengel** [comment]
Just to ask a question - what about build servers?
